packages/ui 타입 export 누락으로 빌드 실패

## 버그 설명
packages/ui에서 필요한 TypeScript 타입들이 export되지 않아 admin/app 앱 생성 후 빌드 시 타입 에러가 발생합니다.

## 재현 방법
1. `pnpm gen:admin` 또는 `pnpm gen:app` (useAuth 포함)으로 앱 생성
2. `pnpm --filter {app-name} build:staging` 실행
3. TypeScript 컴파일 에러 발생

## 에러 메시지

### 1. UseToastReturn 타입 누락
```
src/contexts/ToastContext.tsx(2,25): error TS2305: Module '"@pf-dev/ui/molecules"' has no exported member 'UseToastReturn'.
```

### 2. ToastFunction 타입 누락
```
src/contexts/ToastContext.tsx(12,17): error TS4058: Return type of exported function has or is using name 'ToastFunction' from external module but cannot be named.
```

### 3. DataTable 타입 호환성 문제
```
src/pages/accounts/users/components/UserTable.tsx(26,34): error TS2344: Type 'User' does not satisfy the constraint 'Record<string, unknown>'.
Index signature for type 'string' is missing in type 'User'.
```

## 예상 동작
- 생성된 앱이 타입 에러 없이 빌드 성공

## 실제 동작
- TypeScript 타입 에러로 빌드 실패

## 관련 코드

### 수정 필요한 파일:
1. `packages/ui/src/molecules/Toast/index.ts` - UseToastReturn, ToastFunction export 추가 필요
2. `packages/ui/src/molecules/Toast/useToast.ts` - ToastFunction을 export interface로 변경 필요
3. `packages/ui/src/organisms/DataTable/DataTable.tsx` - 타입 제약 완화 및 가상 컬럼 지원 필요

## 환경
- **TypeScript**: 5.9.3
- **Vite**: 7.3.1
- **React**: 19.2.3

## 해결 방안

### 1. Toast 타입 export
```typescript
// packages/ui/src/molecules/Toast/index.ts
export type { UseToastReturn, ToastFunction } from "./useToast";

// packages/ui/src/molecules/Toast/useToast.ts
export interface ToastFunction { ... }
```

### 2. DataTable 타입 제약 완화
```typescript
// Before
export interface DataTableColumn<T extends Record<string, unknown>>

// After  
export interface DataTableColumn<T> {
  key: keyof T | string;  // 가상 컬럼 지원
  ...
}
```

## 추가 정보
이 버그는 generator 템플릿이 생성한 코드에서 packages/ui의 타입을 사용하려 할 때 발생합니다. 타입이 제대로 export되지 않아 외부 패키지에서 사용할 수 없는 상황입니다.

--- a/.changeset/fix-ui-type-exports.md
+++ b/.changeset/fix-ui-type-exports.md
@@ -1,0 +1,8 @@
+---
+"@pf-dev/ui": patch
+---
+
+packages/ui 타입 export 누락 수정
+
+- Toast: UseToastReturn, ToastFunction 타입 export 추가
+- DataTable: 타입 제약 완화 및 가상 컬럼 지원 (key: keyof T | string)

--- a/packages/ui/src/molecules/Toast/index.ts
+++ b/packages/ui/src/molecules/Toast/index.ts
@@ -9,5 +9,6 @@ export {
 } from "./Toast";
 export { Toaster } from "./Toaster";
 export { useToast } from "./useToast";
+export type { UseToastReturn, ToastFunction } from "./useToast";
 export { toastVariants } from "./variants";
 export type { ToastProps, ToastData } from "./types";

--- a/packages/ui/src/molecules/Toast/useToast.ts
+++ b/packages/ui/src/molecules/Toast/useToast.ts
@@ -18,7 +18,7 @@ export interface ToastData {
 
 type ToastInput = Omit<ToastData, "id">;
 
-interface ToastFunction {
+export interface ToastFunction {
   (input: ToastInput | string): string;
   success: (input: ToastInput | string) => string;
   error: (input: ToastInput | string) => string;

--- a/packages/ui/src/organisms/DataTable/DataTable.tsx
+++ b/packages/ui/src/organisms/DataTable/DataTable.tsx
@@ -5,7 +5,7 @@ import { Checkbox } from "../../atoms/Checkbox";
 import { Button } from "../../atoms/Button";
 
 export interface DataTableColumn<T> {
-  key: keyof T;
+  key: keyof T | string;
   header: string;
   sortable?: boolean;
   render?: (row: T, index: number) => React.ReactNode;
@@ -163,7 +163,7 @@ function DataTablePagination({
   );
 }
 
-function DataTableComponent<T extends Record<string, unknown>>({
+function DataTableComponent<T>({
   data,
   columns,
   selectable = false,
@@ -173,7 +173,7 @@ function DataTableComponent<T extends Record<string, unknown>>({
   className,
 }: DataTableProps<T>) {
   const [selectedRows, setSelectedRows] = useState<Set<number>>(new Set());
-  const [sortColumn, setSortColumn] = useState<keyof T | null>(null);
+  const [sortColumn, setSortColumn] = useState<keyof T | string | null>(null);
   const [sortDirection, setSortDirection] = useState<"asc" | "desc">("asc");
   const [currentPage, setCurrentPage] = useState(1);
 
@@ -206,7 +206,7 @@ function DataTableComponent<T extends Record<string, unknown>>({
     onSelectionChange?.(data.filter((_, i) => newSelected.has(i)));
   };
 
-  const handleSort = (columnKey: keyof T) => {
+  const handleSort = (columnKey: keyof T | string) => {
     if (sortColumn === columnKey) {
       setSortDirection(sortDirection === "asc" ? "desc" : "asc");
     } else {
@@ -217,8 +217,8 @@ function DataTableComponent<T extends Record<string, unknown>>({
 
   const sortedData = [...data].sort((a, b) => {
     if (!sortColumn) return 0;
-    const aVal = a[sortColumn];
-    const bVal = b[sortColumn];
+    const aVal = a[sortColumn as keyof T];
+    const bVal = b[sortColumn as keyof T];
     if (aVal === bVal) return 0;
     if (aVal == null) return 1;
     if (bVal == null) return -1;
@@ -310,7 +310,7 @@ function DataTableComponent<T extends Record<string, unknown>>({
                     >
                       {column.render
                         ? column.render(row, actualIndex)
-                        : String(row[column.key] ?? "")}
+                        : String(row[column.key as keyof T] ?? "")}
                     </td>
                   ))}
                 </tr>
@@ -333,8 +333,6 @@ function DataTableComponent<T extends Record<string, unknown>>({
   );
 }
 
-const DataTable = DataTableComponent as <T extends Record<string, unknown>>(
-  props: DataTableProps<T>
-) => React.ReactElement;
+const DataTable = DataTableComponent as <T>(props: DataTableProps<T>) => React.ReactElement;
 
 export { DataTable, DataTableBulkActionBar, DataTablePagination };

